crates/hash/src/address_impl.rs

```diff
@@ -12,7 +13,17 @@ impl Address for Block {
 
 impl Address for Predicate {
     fn content_address(&self) -> ContentAddress {
-        ContentAddress(hash(self))
+        let Ok(header) = self.encoded_header() else {
+            // Invalid predicates can't be hashed.
+            return ContentAddress([0; 32]);
+        };
+        let mut hasher = <sha2::Sha256 as sha2::Digest>::new();
+        hasher.update(header.fixed_size_header.0);
+        hasher.update(header.lens);
+        for item in self.as_programs() {
+            hasher.update(item);
+        }
+        ContentAddress(hasher.finalize().into())
```

Comment:
I'm actually planning to change our hashing ops to allow byte alignment because of this. I have it in another branch. I think it's probably going to come up for other things so probably better to do that then pad